packages/server/src/unstable-core-do-not-import/stream/sse.ts

```diff
@@ -227,19 +228,38 @@ export function sseStreamConsumer<TData>(opts: {
     },
   });
 
+  let errorLock: Promise<void | 'CANCEL_ALL'> | undefined = undefined;
   eventSource.addEventListener('message', (msg) => {
     stream.controller.enqueue(msg);
   });
   eventSource.addEventListener(SERIALIZED_ERROR_EVENT, (msg) => {
     stream.controller.enqueue(msg);
   });
-  eventSource.addEventListener('error', (cause) => {
+  eventSource.addEventListener('error', async (cause) => {
+    // We prevent more than 1 error handler from waiting at the same time
+    const result = await errorLock;
+    if (result === 'CANCEL_ALL') {
+      return;
+    }
+
+    const resolvable = createResolvable<void | 'CANCEL_ALL'>();
+    errorLock = resolvable.promise;
```

Comment:
we have a utility for this already called `createDeferred`
```suggestion
    const deferred = createDeferred<void | 'CANCEL_ALL'>();
    errorLock = deferred.promise;
```